src/lld/moviebooking/class_diagram.puml

```diff
@@ -1,3 +1,323 @@
+    }
+    class Crew extends Identifier{
+        - crewType: CrewType
+    }
+    enum CrewType{
+        Cameraman
+        Designer
+        Hero
+        Heroine
+        Director
+        Singer
+    }
+    'Todo: Preferred hierarchical approach, should discuss and final best entity design for city state, this takes four level joins
+    class District extends Identifier{
+        cityId: long
+    }
+    class City extends Identifier{
+        stateId: long
+    }
+    class State extends Identifier{
+        countryId: Country
+    }
+    class Country extends Identifier{
+    }
+
+    class Theatre extends Identifier {
+        - theatreGroupId: long
+        - address: Address
+        - screens: List<Screen>
+        - theatreStatus: TheatreStatus
+    }
+    class TheatreGroup extends Identifier{
+        - founder: String
+        - foundedDate: LocalDateTime
+    }
+    class Address{
+        - id: long
+        - addressLine1: String
+        - addressLine2: String
+        - districtId: districtId
+        - gpsLocation: GpsLocation
+        - pinCode: String
+    }
+    class Screen extends Identifier{
+        - seats: List<Seat>
+        - screenStatus: ScreenStatus
+    }
+    enum TheatreStatus{
+        Available
+        NotAvailable
+    }
+    enum ScreenStatus{
+        Available
+        NotAvailable
+    }
+    'Todo: Seat arrangement belongs to the screen style, user can customise it'
+    class Seat extends Identifier{
+        - rowNo: String
+        - colNo: String
+        - seatStatus: SeatStatus
+    }
+    enum SeatStatus{
+        Available,
+        NotAvailable,
+        Locked
+    }
+    enum SeatCategory{
+        Normal
+        Premium
+    }
+    class Show extends Identifier{
+        - showTime: LocalDateTime
+        - screenId: long
+    }
+}
+
+package com.movie_booking.entity.mapping{
+    class Booking {
+        - bookingId: long
+        - showId: long
+        - seatId: long
+        - userId: long
+        - movieId: long
+        - theatreId: long
+        - screenId: long
+        - bookingTime: LocalDateTime
+        - paymentId: long
+        - bookingStatus: BookingStatus
+    }
+    enum BookingStatus{
+        Booked,
+        Cancelled
+    }
+    class SeatDTO{
+        seat: Seat,
+        showId: long
+    }
+    class TheatreShowDTO{
+        theatre: Theatre,
+        shows: List<Show>
+    }
+    class MovieDTO{
+        movie: Movie
+        supportedLanguagesInCity: List<City>
+        cityWiseDetails:
+    }
+    class cityWiseDetails{
+        cityId: long
+        releaseTime: LocalDateTime
+    }
+}
+package com.movie_booking.dao.main{
+    interface ReadDao<T>{
+        + findById(id: long): T
+        + findAll(): List<T>
+    }
+    interface WriteDao<T>{
+        + create(entity: T): T
+        + update(id: long, entity: T): T
+        + delete(id: long): boolean
+    }
+    class DBReadUserDao<User> implements ReadDao{
+        - connection: Connection
```

Comment:
can connection be moved to DBDao a abstract class?

Reply:
done and moved to common package